docs/did/asset-did-spec.md

```diff
@@ -0,0 +1,107 @@
+
+### Version History
+
+- **v1.0 - July.04 2022**: Initial definition and RFC for the new method
+
+---
+
+## Abstract
+
+This document defines the Asset DID Method that conforms to the [DID Core W3C Spec][did-core-spec].
+By the term *asset*, the authors refer to both an entire asset (or token) class (e.g., the entire Bitcoin fungible asset class) as well as a specific instance within that class (e.g., a specific NFT - the instance - within an NFT collection - the asset class).
+
+Hence, an Asset Decentralized Identifier (DID) is a string uniquely identifying assets on any existing and future blockchain.
+
+## Method Syntax
+
+### DID Scheme and Identifiers
+
+The Asset DID method is identified by the `asset` scheme, as in the following example: `did:asset:bip122:000000000019d6689c085ae165831e93.slip44:0`.
+
+Asset DIDs are composed of the following elements:
+
+```
+asset-did                   = "did:asset:" + <modified-caip-19-asset-id>
+modified-caip-19-asset-id   = <caip-2-chain-id> + "." + <asset-namespace> + ":" + <asset-reference> + [":" + <asset-id>]
+caip-2-chain-id             = <chain-namespace> + ":" + <chain-reference>
+chain-namespace             = [-a-z0-9]{3,8}
+chain-reference             = [-a-zA-Z0-9]{1,32}
+asset-namespace             = [-a-z0-9]{3,8}
+asset-reference             = [-a-zA-Z0-9]{1,64}
+asset-id                    = [-a-zA-Z0-9]{1,78}
+```
+
+The `caip-2-chain-id` is a CAIP (Chain Agnostic Improvement Proposal) chain identifier as described in the [CAIP-2 Blockchain ID Specification][caip-2-spec].
+The `modified-caip-19-asset-id` is derived from a CAIP asset identifier as described in the [CAIP-19 Asset Type and Asset ID Specification][caip-19-spec], with the `/` symbol separating the chain identifier from the asset identifier being replaced with the `.` symbol, and the optional `/` symbol preceding the `asset-id` component being replaced by the `:` symbol.
+This is required to render the resulting string a valid URI according to the [DID Core Specification][did-core-spec].
+
+Chain and asset namespaces are defined with Improvement Proposals and are stored under the [CAIP repository][caip-repo].
+For instance, the namespaces for [EIP155][caip-3-spec] (Ethereum-based) and [BIP122][caip-4-spec] (Bitcoin-based) chains as well as for [SLIP44][caip-20-spec] (native fungible tokens), [ERC20][caip-21-spec] (smart contract based tokens), and [ERC721][caip-22-spec] (NFTs) assets have already been defined.
+
+## Method Operations
+
+### Create an Asset DID
+
+A new Asset DID creation is automatically performed whenever a new asset class is deployed or a new asset instance for a given class minted.
+Because of the chain agnostic nature of Asset DIDs, this process can take place on any blockchain that is identifiable by a [CAIP-2 Identifier][caip-2-spec].
+
+### Resolve an Asset DID
+
+The information required to resolve an Asset DID is the following:
```

Comment:
resolve what from an Asset DID? a DID document?

Reply:
That would be one way of looking at it, sure. My idea was to define a "skeleton" for the DIDDocument that we can then expand in the future as we wish, for instance by linking assets on multiple blockchains together with the `alsoKnownAs` method. Also, although I might agree having a property on the type of asset (fungible/non-fungible) might bring some value, I am not sure what value would the other properties such as asset-namespace bring. This would just mean maintenance burden for us, and also the definition of all the possible values these properties can have in the JSON-LD context file.